src/Service/UserService.php

```diff
@@ -8,10 +8,77 @@
 
 namespace App\Service;
 
+use Doctrine\Common\Persistence\ObjectManager;
 use Doctrine\ORM\EntityManager;
 use App\Entity\User;
 
 class UserService
 {
+    private $manager;
+    private $templating;
+
+
+    /**
+     * UserService constructor.
+     * @param ObjectManager $manager
+     * @param \Twig_Environment $templating
+     */
+    public function __construct(ObjectManager $manager, \Twig_Environment $templating)
+    {
+        $this->manager = $manager;
+        $this->templating = $templating;
+    }
+
+    /**
+     * @param string $token
+     * @return array
+     * @throws \Exception
+     */
+    public function verifyToken(string $token) {
+        $user = $this->manager->getRepository(User::class)
+            ->findOneBy(array('email_token' => $token));
+
+        if ($user && !$user->getVerified()) {
+            $user->setVerified(true);
+            $this->manager->flush();
+            return ["isVerified" => true,
+                "login" => $user->getEmail()];
+        } else if($user && $user->getVerified()) {
+            throw new \Exception('Something went wrong!');
+        } else if(!$user) {
+            throw new \Exception('Something went wrong!');
+        } else {
+            throw new \Exception('Something went wrong!');
```

Comment:
Not important right now but we should have separate exceptions to be able to know exactly what happened !